README
CONTENTICIOUS

Contenticious is a simple way to build a pretty simple website from Markdown
content. It includes a helper script which will create example pages and
directories for you to get you started in no time. With one single command
Contenticious will create static HTML ready for upload. It's also possible
to mount Contenticious in existing Mojolicious web apps.

See http://memowe.github.com/contenticious for an introduction.


PREREQUISITES

    - perl 5.10.1
    - File::Copy::Recursive 0.38
    - Mojolicious 4.90
    - Mojolicious::Plugin::Subdispatch 0.04
    - Mojolicious::Plugin::RelativeUrlFor 0.051
    - Text::Markdown 1.000031


INSTALLATION

This distribution is available on CPAN. You can install it like any other
CPAN module via

    $ cpan Contenticious

or

    $ cpanm Contenticious

See their documentation for details. To install it manually,
download the dist tarball or clone the git repository
and execute the following standard procedure:

    $ cpanm --notest .
    $ perl Makefile.PL
    $ make
    $ make test
    $ make install

It's possible to use this module from inside of its dist directory without
installation.


DOCUMENTATION

After installation you can access the main documentation with this command:

    $ perldoc Contenticious

It's possible to view the same file without installing Contenticious:
http://search.cpan.org/dist/Contenticious/lib/Contenticious.pod


AUTHOR AND LICENSE

Copyright (c) Mirko Westermeier, <mail@memowe.de>

Credits:
- Maxim Vuets, <maxim.vuets@gmail.com>
- Stephan Jauernick, <stephan@stejau.de>

Thank you for your contributions!

Published under the MIT license. See MIT-LICENSE.


Changes
Revision history for perl module Contenticious

0.393   2018-04-10
    - Bugfix release because of a non-stated dependency

0.392   2018-04-09

    - generate boilerplate from file system
    - use M::P::PODViewer (M::P::PODRenderer is deprecated since M 8.05)

0.391   2017-01-31

    - keep up with changes in Mojo::Home (@kiwiroy)

0.39    2016-09-29

    - fix file/directory handling in the presence of spaces

0.38    2016-07-21

    - fix Mojolicious 7.0 compatibility

0.35    2015-02-23

    - updated prerequisites

0.333   2013-11-28

    - changed DOM tests to use attr instead of the deprecated attrs (M. 4.20)

0.332   2012-12-11

    - forgot to add the latest M::P::RelativeUrlFor changes to deps - sorry!

0.331   2012-12-09

    - added a dummy base url for M::P::Subdispatch to correctly generate
      relative urls

0.33    2012-10-31

    - disabled RelativeUrlFor's replace_url_for and switched to manual
      IMPORTANT: maybe you have to replace url_for with rel_url_for in
                 your existing inflated templates!

0.328   2012-09-13

    - temporarily disabled the (not so important) mount test
      because FindBin dies in Mojo::Server

0.327   2012-09-11

    - replaced Mojo::Command by Mojolicious::Command (Joel Berger)

0.326   2012-08-30

    - removed MOJO_APP usage

0.325   2012-04-29

    - removed a small incompatibility in content.html.ep (%== instead of b)

0.324   2012-03-27

    - increased Mojolicious version because of a small inflate command bug

0.323   2012-03-27

    - updated Mojolicious::Renderer->default_template_class
      and Mojolicious::Static->root calls
    - added hypnotoad configuration documentation

0.322   2012-03-08

    - a better Makefile.PM with MIN_PERL_VERSION and META informations

0.321   2012-03-06

    - fixed a bug in t/06_script.t (tinita, stephan48)
      cpantesters reports should be much "greener" now

0.32    2012-03-05

    - updated README (Contenticious is on CPAN now)
    - added morbo support

0.311   2012-03-03

    - changed nothing important, just needed to bump the version number
      to be able to upload tarball to CPAN (my fault)

0.31    2012-02-22

    - fixed windows compatibility for the script test (stephan48)

0.3     2012-02-21

    - really installable app with a standalone contenticious script

0.2     2012-01-12

    - rewrite with Makefile.pm and everything

0.1     2010-01-15

    - first version


Makefile.PL
use 5.010001;

use strict;
use warnings;

use ExtUtils::MakeMaker;
use File::ShareDir::Install;

install_share dist => 'share';

WriteMakefile(
    NAME                => 'Contenticious',
    VERSION_FROM        => 'lib/Contenticious.pm',
    ABSTRACT            => 'build web sites from markdown files',
    AUTHOR              => 'Mirko Westermeier <mail@memowe.de>',
    LICENSE             => 'mit',
    META_MERGE          => {
        resources   => {
            homepage    => 'http://memowe.github.com/contenticious/',
            bugtracker  => 'http://github.com/memowe/contenticious/issues',
            repository  => 'http://github.com/memowe/contenticious',
        },
    },
    MIN_PERL_VERSION    => '5.10.1',
    PREREQ_PM           => {

        # core modules since 5.10
        'Carp'                                  => '1.08',
        'Cwd'                                   => '3.62',
        'File::Basename'                        => '2.76',
        'File::Copy'                            => '2.11',
        'File::Path'                            => '2.04',
        'File::Spec'                            => '3.62',
        'FindBin'                               => '1.49',
        'List::Util'                            => '1.19',

        # non-trivial dependencies
        'File::Copy::Recursive'                 => '0.38',
        'File::Share'                           => '0.25',
        'Mojolicious'                           => '8.05',
        'Mojolicious::Plugin::AssetPack'        => '1.45',
        'Mojolicious::Plugin::PODViewer'        => '0.004',
        'Mojolicious::Plugin::RelativeUrlFor'   => '0.052',
        'Mojolicious::Plugin::Subdispatch'      => '0.04',
        'Text::Markdown'                        => '1.000031',
    },
    CONFIGURE_REQUIRES  => {
        'ExtUtils::MakeMaker'                   => 0,
        'File::ShareDir::Install'               => '0.13',
    },
    EXE_FILES           => ['contenticious'],
    test                => {TESTS => 't/*.t'}
);

package MY;
use File::ShareDir::Install 'postamble';
